.github/workflows/ci.yml

```
@@ -1,23 +1,60 @@
 name: Continuous Integration
 on: 
   push:
+   # branches:    
+    #- main
+   # - 'Mod/**'
     paths-ignore:
     - '**/README.md'
-
+env:
+      # Use docker.io for Docker Hub if empty
+      REGISTRY: docker.io
+      # github.repository as <account>/<repo>
+      IMAGE_NAME: ${{ github.repository }}
 jobs:
-  build:
-    name: Build and test
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@v3
-    - name: Run Snyk to check for vulnerabilities
-      uses: snyk/actions/python@master
-      continue-on-error: true
-      env:
-        SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
-        
-    - run: docker build --target test --tag todo-app:test .
-      name: build docker test image
-      
-    - run: docker run --env-file .env.test -p 8100:8100 todo-app:test
-      name: run docker test image
\ No newline at end of file
+      test:
+          name: test
+          runs-on: ubuntu-latest
+          steps:
+              - uses: actions/checkout@v3
+              - uses: snyk/actions/python@master
+#              - name: Run Snyk to check for vulnerabilities
+                continue-on-error: true
+                env:
+                  SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+  
+              - run: docker build --target test --tag todo-app:test .
+              - run: docker run --env-file .env.test -p 5000:5000 todo-app:test
+  
+      deploy:
+          name: deploy
+          needs: test
+          runs-on: ubuntu-latest
+          steps:
+              - run: echo Publishing!
+              - uses: actions/checkout@v3
+              - uses: docker/setup-buildx-action@v2
+#              - name: login to docker
+              - uses: docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a
+                with:
+                  username: ${{ secrets.DOCKER_USERNAME }}
+                  password: ${{ secrets.DOCKER_PASSWORD }}
+#              - name: Extract metadata (tags, labels) for Docker
+#                id: meta
+#              - uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+#                with:
+#                  #images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+#                  images: lvarnham/smellybeagle:latest
+              
+              - name: Build and push Docker image
+                uses: docker/build-push-action@v2
+                with:
+                  #  context: "{{defaultContext}}"
+                    push: true
+                    tags: lvarnham/smellybeagle:latest
+                    labels: production
+
+              - run: docker build --target production --tag lvarnham/smellybeagle:latest .
+              - run: docker push lvarnham/smellybeagle:latest
+              - run: curl -dH -X POST "https://\$smellybeagle:p1DcTzc2mkukhaL8261Zemq7CdLoDNe97AyagKjwqtBlw5NTN98Q1Tt2vu3p@smellybeagle.scm.azurewebsites.net/api/registry/webhook"
```

Panic! This is a sensitive password, so we should rotate it immediately, and then replace it with a secret here to let us use it securely.

In order to rotate it, the best process is to:
* Visit your app service in Azure
* Enter the "Deployment Center" in the left hand menu
* Move to the "FTPS credentials" tab
* Click the "Reset" button next to the "Password" field under "Application Scope"

In order to use it securely here, the best approach is:
* Put the raw full webhook into a secret within your GitHub repo settings (i.e. it _doesn't_ have the `\` before the `$`)
* Update this step to use that from an environment variable (this conversion then handles the escaping):
```suggestion
              - env:
                  WEBHOOK: "${{ secrets.WEBHOOK }}"
                run: curl -dH -X POST $WEBHOOK
```